app/models/king.rb

```diff
@@ -1,7 +1,68 @@
+# rubocop:disable Metrics/AbcSize
 class King < Piece
   def move_legal?(to_row, to_col)
     row_diff = (row - to_row).abs
     col_diff = (col - to_col).abs
     row_diff <= 1 && col_diff <= 1
+    # add castling moves to the 'whitelist', then verify the validity of the castle
```

Comment:
Does `move_legal?` need to ensure you can't move yourself into check?  Or is that handled elsewhere?